hack/build-image-docker.sh

```diff
@@ -0,0 +1,215 @@
+build_with_docker_buildx() {
+  echo "Building multi-arch image $image ..."
+  
+  # Create platforms string for buildx
+  platforms=""
+  for arch in $architectures; do
+    if [ -n "$platforms" ]; then
+      platforms="$platforms,linux/$arch"
+    else
+      platforms="linux/$arch"
+    fi
+  done
+  
+  # For push builds, use multi-platform; for local builds, build per arch
+  if [ -z "${DRY_RUN:-}" ]; then
+    # Building for push - use multi-platform with --push
+    docker buildx build \
+      --file Dockerfile \
+      --tag "$image" \
+      --platform "$platforms" \
+      --build-arg "TARGETOS=linux" \
+      --push \
+      .
+  else
+    # For local/dry-run builds, build each architecture separately with --load
+    for arch in $architectures; do
+      fullTag="$image-$arch"
+      echo "Building $fullTag ..."
+      docker buildx build \
+        --file Dockerfile \
+        --tag "$fullTag" \
+        --platform "linux/$arch" \
+        --build-arg "TARGETOS=linux" \
+        --build-arg "TARGETARCH=$arch" \
+        --load \
+        .
+    done
+    # Tag the first architecture as the main image for local use
+    first_arch=$(echo $architectures | cut -d' ' -f1)
+    docker tag "$image-$first_arch" "$image"
+  fi
+}
+
+# Function to build images with regular docker (single arch only)
+build_with_docker() {
+  # Use only the first architecture for regular docker
+  arch=$(echo $architectures | cut -d' ' -f1)
+  fullTag="$image-$arch"
+  
+  echo "Building single-arch image $fullTag (docker without buildx) ..."
+  docker build \
+    --file Dockerfile \
+    --tag "$fullTag" \
+    --platform "linux/$arch" \
+    --build-arg "TARGETOS=linux" \
+    --build-arg "TARGETARCH=$arch" \
+    .
+    
+  # Tag it as the main image too
+  docker tag "$fullTag" "$image"
+}
+
+# Build images based on available docker features
+if [ "$DOCKER_BUILDX" = true ]; then
+  build_with_docker_buildx
+else
+  build_with_docker
+fi
+
+# Additionally to an image tagged with the Git tag, we also
+# release images tagged with the current branch name, which
+# is somewhere between a blanket "latest" tag and a specific
+# tag.
+if [ -n "$branchName" ] && [ -z "${PULL_NUMBER:-}" ]; then
+  branchImage="$repository:$branchName"
+
+  if [ "$DOCKER_BUILDX" = true ]; then
+    echo "Tagging multi-arch image as $branchImage ..."
+    docker tag "$image" "$branchImage"
+  else
+    echo "Tagging single-arch image as $branchImage ..."
+    docker tag "$image" "$branchImage"
+  fi
+fi
+
+# push images, except in dry runs
+if [ -z "${DRY_RUN:-}" ]; then
+  echo "Logging into GHCR ..."
+  
+  if [ "$DOCKER_BUILDX" = true ]; then
+    # buildx with --push already pushed during build
+    echo "Images already pushed during buildx build"
+  else
+    # Regular docker - need to login and push
+    if [ -n "${KCP_GHCR_USERNAME:-}" ] && [ -n "${KCP_GHCR_PASSWORD:-}" ]; then
+      echo "$KCP_GHCR_PASSWORD" | docker login ghcr.io -u "$KCP_GHCR_USERNAME" --password-stdin
```

Comment:
Similar thing here: If this isn't meant for CI, let's just run through these automated login steps?